packages/node_modules/@webex/webex-core/src/lib/services/services.js

```diff
@@ -684,6 +688,74 @@ const Services = WebexPlugin.extend({
       .then(({body}) => this._formatReceivedHostmap(body));
   },
 
+  /**
+   * Initialize the discovery services and the whitelisted services.
+   *
+   * @returns {void}
+   */
+  initConfig() {
+    // Get the catalog and destructure the services config.
+    const catalog = this._getCatalog();
+    const {services} = this.webex.config;
+
+    // Validate that the services configuration exists.
+    if (services) {
+      // Check for discovery services.
+      if (services.discovery) {
+        // Format the discovery configuration into an injectable array.
+        const formattedDiscoveryServices = Object.keys(services.discovery)
+          .map((key) => ({
+            name: key,
+            defaultUrl: services.discovery[key]
+          }));
+
+        // Inject formatted services into services catalog.
+        catalog.updateServiceUrls('discovery', formattedDiscoveryServices);
+      }
+
+      // Check for allowed host domains.
+      if (services.allowedDomains) {
+        // Store the allowed domains as a property of the catalog.
+        catalog.setAllowedDomains(services.allowedDomains);
+      }
+
+      // Set `validateDomains` property to match configuration
+      this.validateDomains = services.validateDomains;
+    }
+  },
+
+  /**
+   * Make the initial requests to collect the root catalogs.
+   *
+   * @returns {Promise<void, Error>} - Errors if the token is unavailable.
+   */
+  initServiceCatalogs() {
+    this.logger.info('services: initializing initial service catalogs');
+
+    // Destructure the credentials plugin.
+    const {credentials} = this.webex;
+
+    // Init a promise chain to catch errors.
+    return Promise.resolve()
+      // Get the user's OrgId.
+      .then(() => credentials.getOrgId())
```

Comment:
Should you have a `catch` since `getOrgId` might throw an error? or do you want to just let it propagate?

Reply:
also what if credentials are not set yet? it won't be able to get an org id

Reply:
@aimeex3, I've made this a pre-requisite for the initial event. This method will no longer be called automatically if there isn't a valid supertoken.

@lalli-flores, now that there's a required check before this is automatically triggered, in the case of this class method, the mentioned method will always return an orgId.

Reply:
@InteractiveTimmy wanted to call it out. afaik guest anonymous tokens don't technically have an `orgId`, however it still does return a an `orgId`. i dunno if it's a placeholder, or real, or what, but something i wanted to call attention to